src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/domainevents/DomainEventDescriber.kt

```diff
@@ -121,9 +123,22 @@ class DomainEventDescriber(
     return event.describe { "A placement was not made for the placement request.$failureReason" }
   }
 
+  @SuppressWarnings("TooGenericExceptionThrown")
   private fun buildBookingCancelledDescription(domainEventSummary: DomainEventSummary): String? {
     val event = domainEventService.getBookingCancelledEvent(domainEventSummary.id())
-    return event.describe { "The placement was cancelled. The reason was: '${it.eventDetails.cancellationReason}'" }
+    return event.describe { data ->
+      val booking = bookingRepository.findByIdOrNull(data.eventDetails.bookingId)
+        ?: throw RuntimeException("Booking ID ${data.eventDetails.bookingId} with cancellation not found")
+      if (booking.cancellations.count() != 1) {
+        throw RuntimeException("Booking ID ${data.eventDetails.bookingId} does not have one cancellation")
+      }
+      val cancellation = booking.cancellations.first()
+      val otherReasonText = if (cancellation.reason.name == "Other" && !cancellation.otherReason.isNullOrEmpty()) ": ${cancellation.otherReason}." else ""
```

Comment:
I _think_ it may be better to add the UUID for the 'other' reason to CancellationReasonRepository.Constants and use that to identify the 'other' reason. Neither approach is ideal but we've used that approach elsewhere.